### PR TITLE
Jr.Devjobs: Improve regex, tabbing, and add tests

### PR DESCRIPTION
**Pull request by HunterMeyer**
###### What does your Pull Request do (check all that apply)?

- [X] Improvement
    - [X] Enhancement: **`Jr.DevJobs: Improve regex, tabbing, and add tests`**

###### Description of changes

Use 4-space soft tabs.
Update regex matching
Use `startend` triggers
Add more tests

###### People to notify (@mention interested parties)
@moollaza 

---

Instant Answer Page: https://duck.co/ia/view/jr_dev_jobs

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by edgesince84** [comment]
Thanks for the PR, Hunter!  We'll get it reviewed soon 👍

**Comment by moollaza** [comment]
@HunterMeyer thanks a lot! Deploying to Beta for testing...

File: lib/DDG/Spice/JrDevJobs.pm [hunk 1]
```diff
@@ -1,18 +1,17 @@
...
 
 spice to => 'https://www.jrdevjobs.com/jobs.json?query=$1';
 spice is_cached => 1;
 spice wrap_jsonp_callback => 1;
 
 handle remainder => sub {
-   s/(jobs)//;
-   s/(job)//;
-   s/(developer)//;
+    s/(jobs?|developer)//i;
```
Comment (moollaza) [inline lib/DDG/Spice/JrDevJobs.pm]: Let's anchor these words so we don't replace a partial word: `/\b(jobs?|developer)\b/`

File: lib/DDG/Spice/JrDevJobs.pm [hunk 2]
```diff
@@ -1,18 +1,17 @@
...
```
Comment (moollaza) [inline lib/DDG/Spice/JrDevJobs.pm]: Let's add `use Text::Trim` to the top, and strip and trailing spaces before sending it to your API:

`return trim($_) if $_;`

File: t/JrDevJobs.t [hunk 3]
```diff
@@ -4,6 +4,22 @@ use warnings;
...
+        'DDG::Spice::JrDevJobs'
+    ],
+    'jrdevjobs ruby' => test_spice(
+        '/js/spice/jr_dev_jobs/ruby',
+        call_type => 'include',
+        caller => 'DDG::Spice::JrDevJobs',
+        is_cached => 1
+    ),
+    'web developer jrdevjobs' => test_spice(
+        '/js/spice/jr_dev_jobs/web%20',
```
Comment (moollaza) [inline t/JrDevJobs.t]: with `trim` you can drop the `%20` because there won't be any leading/trailing spaces 👍

**Comment by moollaza** [comment]
@HunterMeyer looking good, left a few comments and then we can merge 👍

**Comment by HunterMeyer** [comment]
@moollaza thanks again for your excellent feedback! I've updated the PR.

**Comment by moollaza** [comment]
@HunterMeyer thanks! Deploying to Beta...

**Comment by moollaza** [comment]
Thanks! This LGTM :+1:

**Comment by edgesince84** [comment]
hey @HunterMeyer, thank you for your contribution to DuckDuckHack, we really appreciate it! I'm reaching out to let you know that we recently launched a new forum with the initiative of improving programming-related searches, and I hope you'll [join us](https://forum.duckduckhack.com/)!

There are a lot of important tasks up for grabs that we'd like your help with, so feel free to and [jump in](https://forum.duckduckhack.com/t/duckduckhack-programming-mission-overview/53).

**Pull request merged**